### ISSUE TITLE
Question about sentence classification experiment

### ISSUE THREAD

**Issue by Chaiyut1002** [GaryYufei/PromDA#4]
Sorry to bother you, I'm very interested in your work and have read most of the code of the whole project. I have replicated your sentence classification experiment, but the results are not good.  Did I do something wrong? The detailed results are as follows.

seed             baseline-shot-10        PromDA-shot-10
18          |         76.66                  |         78.14
45          |         62.66                  |          69.3 
76          |         75.56                  |         69.96
256        |         70.51                  |         73.37
512        |         68.15                  |         69.69
Avg        |         69.09                 |          73.03

**Comment by GaryYufei** [comment]
Have you used my pre trained checkpoint? Can I have a look at one of your
training log

**Comment by Chaiyut1002** [comment]
Yes, I've used pre-trained checkpoint downloaded from google drive. This is the result with the seed set to 10. Do I need to send you the full folder?

[sst2_512_shot_10_tag_keyword_two_prefix_1000_step.txt](https://github.com/GaryYufei/PromDA/files/8495015/sst2_512_shot_10_tag_keyword_two_prefix_1000_step.txt)

**Comment by GaryYufei** [comment]
> Yes, I've used pre-trained checkpoint downloaded from google drive. This is the result with the seed set to 10. Do I need to send you the full folder?
> 
> [sst2_512_shot_10_tag_keyword_two_prefix_1000_step.txt](https://github.com/GaryYufei/PromDA/files/8495015/sst2_512_shot_10_tag_keyword_two_prefix_1000_step.txt)
<img width="567" alt="Screen Shot 2022-04-15 at 6 59 38 pm" src="https://user-images.githubusercontent.com/8288965/163550388-81bfdf0e-77eb-437d-ae08-b601af8ab04b.png">

This is unlikely to happen (normally high success ratio in sst2 data). Must be something wrong here. Could you look into this? I will also check the code during the weekend.

**Comment by Chaiyut1002** [comment]
Maybe it is caused by get_sentence_classification_output() in train_data_augmentation.py? It seems to only keep sentences where the first word is the label. But there is a problem that the first word of the label is capitalized (Positve, Negative), which leads to no generated sentences in nlg_model_mix_output_part1.txt. 

If I modify line 75 in train_data_aumentation.py: 
     label = words[0].lower()
There will be some sentences in nlg_model_mix_output_part1.txt.

![before](https://user-images.githubusercontent.com/90961172/163560515-07f53932-24e4-4ccc-bf99-d38ffb64c85d.png)
![after](https://user-images.githubusercontent.com/90961172/163560523-76e3eb62-cf6e-4c0b-934d-1ada8d208d10.png)

**Comment by GaryYufei** [comment]
> Maybe it is caused by get_sentence_classification_output() in train_data_augmentation.py? It seems to only keep sentences where the first word is the label. But there is a problem that the first word of the label is capitalized (Positve, Negative), which leads to no generated sentences in nlg_model_mix_output_part1.txt.
> 
> If I modify line 75 in train_data_aumentation.py: label = words[0].lower() There will be some sentences in nlg_model_mix_output_part1.txt.
> 
> ![before](https://user-images.githubusercontent.com/90961172/163560515-07f53932-24e4-4ccc-bf99-d38ffb64c85d.png) ![after](https://user-images.githubusercontent.com/90961172/163560523-76e3eb62-cf6e-4c0b-934d-1ada8d208d10.png)

Thanks for the update. I will look into this issue.

**Comment by GaryYufei** [comment]
Hi @Chaiyut1002,  Could you try to run the code again? I have fixed the issue -- Due to a small bug, the output view prompt is never trained. So it produces wired outputs. I believe the current version should solve the problem

**Comment by Chaiyut1002** [comment]
I tried it, but it still doesn't work properly. Could you upload the version without problems to google drive?  training log is as follows.
[sst2_18_shot_10_tag_keyword_two_prefix_1000_step.txt](https://github.com/GaryYufei/PromDA/files/8501118/sst2_18_shot_10_tag_keyword_two_prefix_1000_step.txt)

**Comment by GaryYufei** [comment]
OK. I fixed that again. I just realized my fix last night did not fixed the issue. I think this time should be fine. This time, when training the DA model, could you print `batch['task_index']` to see if it has 0 and 1 mixed. It should be OK if it has both 0 and 1 in each batch.

**Comment by GaryYufei** [comment]
> I tried it, but it still doesn't work properly. Could you upload the version without problems to google drive? training log is as follows. [sst2_18_shot_10_tag_keyword_two_prefix_1000_step.txt](https://github.com/GaryYufei/PromDA/files/8501118/sst2_18_shot_10_tag_keyword_two_prefix_1000_step.txt)

From your training log, one of the views was working well (success ratio is ~98%). Both views are initialised from the same prompt parameters. So it should work well. The problem is that, in my previous version, `_C.prefix_set_number` is set to one in order to load my pre-trained model, but I did not set it back to `old_prefix_set_number`. It causes the `task_index` in data loader always to be 0 (see https://github.com/GaryYufei/PromDA/blob/8e711f3231673b21f678c89282f1778dd89b3ca2/pre_train_dataset.py#L400). Now it should work well.

**Comment by Chaiyut1002** [comment]
Thanks a lot! It works well.

**Comment by GaryYufei** [comment]
> Thanks a lot! It works well.

Great to know it! 

It will be great if you could post your replication results here so other people could also know them.

**Comment by Chaiyut1002** [comment]
> > Thanks a lot! It works well.
> 
> Great to know it!
> 
> It will be great if you could post your replication results here so other people could also know them.

I'm glad to do that.
[sst2_45_shot_10_tag_keyword_two_prefix_1000_step.txt](https://github.com/GaryYufei/PromDA/files/8501578/sst2_45_shot_10_tag_keyword_two_prefix_1000_step.txt)
